cmu_graphics/cmu_graphics.py

```diff
@@ -208,29 +208,95 @@ def __iter__(self): return iter(self._shape)
     def __len__(self): return len(self._shape._shapes)
 
 class Sound(object):
+    initialized = False
+    number_of_sounds = 0
     def __init__(self, url):
-        self.sound = slNewSound(url)
-
-    def play(self, **kwargs):
-        default_kwargs = {'loop': False, 'restart': False}
-
-        for keyword in kwargs:
-            english_keyword = toEnglish(keyword, 'shape-attr')
-            if english_keyword not in default_kwargs:
-                raise Exception("TypeError: %s.%s() got an unexpected keyword argument '%s'" % (t('Sound'), t('play'), keyword))
-            default_kwargs[english_keyword] = kwargs[keyword]
+        try:
+            if not pygame.mixer.get_init():
+                pygame.mixer.init()
+                pygame.mixer.set_num_channels(1)
+                Sound.initialized = True
+        except Exception as e:
+            print('Error initializing sound module:', e)
+            print('Game will continue without sound.')
+
+        if Sound.initialized:
+            if not isinstance(url, str):
+                callSpec = '{className}.{attr}'.format(className=t('Sound'), attr=t('url'))
+                err = t(
+                        '{{error}}: {{callSpec}} should be {{typeName}} (but {{value}} is of type {{valueType}})',
+                        {'error': t('TypeError'), 'callSpec': callSpec, 'typeName': 'string', 'value': repr(url), 'valueType': type(url).__name__}
+                        )
+                raise Exception(err)
+
+            Sound.number_of_sounds += 1
+            if pygame.mixer.get_num_channels()==Sound.number_of_sounds:
+                pygame.mixer.set_num_channels(Sound.number_of_sounds * 2)
+
+            if url.startswith('file://'):
+                url = url.split('//')[-1]
+            if url.startswith('http'):
+                for i in range(10):
+                    try:
+                        response = webrequest.get(url)
+                        self.sound = io.BytesIO(response.read())
+                    except:
+                        if i<9:
+                            continue
+                        else:
+                            raise Exception('Failed to load sound data')
+                    break
+            elif hasattr(__main__, '__file__'):
+                self.sound = os.path.abspath(__main__.__file__ + '/../' + url)
```

Comment:
import __main__ : line 1072
There is two ways for a student to run cmu_graphics.
1. from a file, in which case when the program is run `__main__.__file__` is automatically defined by python as the file the program was started with. I think then it makes sense to have any media import be relative to that file. 
2. from the repl, in which case `__main__.__file__` isn't defined, and we default to have media imports be relative to the cwd.

We indeed don't do something analogous for images, yet, but I think we should do it as it is more logical in this way and to be consistent. Analogous logic for images can be found in progress in 

- #77

(ps: it would be nice getting comment/feedback on the later :)